src/gui/sdl_mapper.cpp

```diff
@@ -2787,17 +2781,68 @@ static bool load_binds_from_file(const std::string_view mapperfile_path,
 
 void MAPPER_CheckEvent(SDL_Event *event)
 {
-	for (auto &group : bindgroups)
-		if (group->CheckEvent(event))
+	switch (event->type) {
+	case SDL_CONTROLLERDEVICEADDED:
+	case SDL_CONTROLLERDEVICEREMOVED:
+	case SDL_JOYDEVICEREMOVED:
+	case SDL_JOYDEVICEADDED:
+		MAPPER_HandleJoyDeviceEvent(
+		        reinterpret_cast<SDL_JoyDeviceEvent*>(event));
```

Comment:
This reinterpret_cast causes a PVS Studio warning [V641](https://pvs-studio.com/en/docs/warnings/v641/): _The size of the '& event' buffer is not a multiple of the element size of the type 'SDL_JoyDeviceEvent'._

We need a follow-up PR to fix this.